src/services/refactors/convertExport.ts

```diff
@@ -0,0 +1,168 @@
+/* @internal */
+namespace ts.refactor {
+    const refactorName = "Convert export";
+    const actionNameDefaultToNamed = "Convert default export to named export";
+    const actionNameNamedToDefault = "Convert named export to default export";
+    registerRefactor(refactorName, {
+        getAvailableActions(context): ApplicableRefactorInfo[] | undefined {
+            const info = getInfo(context);
+            if (!info) return undefined;
+            const description = info.wasDefault ? Diagnostics.Convert_default_export_to_named_export.message : Diagnostics.Convert_named_export_to_default_export.message;
+            const actionName = info.wasDefault ? actionNameDefaultToNamed : actionNameNamedToDefault;
+            return [{ name: refactorName, description, actions: [{ name: actionName, description }] }];
+        },
+        getEditsForAction(context, actionName): RefactorEditInfo {
+            Debug.assert(actionName === actionNameDefaultToNamed || actionName === actionNameNamedToDefault);
+            const edits = textChanges.ChangeTracker.with(context, t => doChange(context.file, context.program, Debug.assertDefined(getInfo(context)), t, context.cancellationToken));
+            return { edits, renameFilename: undefined, renameLocation: undefined };
+        },
+    });
+
+    type ExportToConvert = (FunctionDeclaration | ClassDeclaration | InterfaceDeclaration | EnumDeclaration | NamespaceDeclaration) & { readonly name: Identifier };
+    interface Info {
+        readonly exportNode: ExportToConvert;
+        readonly wasDefault: boolean;
+    }
+
+    function getInfo(context: RefactorContext): Info | undefined {
+        const { file } = context;
+        const span = getRefactorContextSpan(context);
+        const token = getTokenAtPosition(file, span.start, /*includeJsDocComment*/ false);
+        const exportNode = getParentNodeInSpan(token, file, span);
+        if (!exportNode || exportNode.parent !== file) return undefined;
+
+        const flags = getModifierFlags(exportNode);
+        const wasDefault = !!(flags & ModifierFlags.Default);
+        // If source file already has a default export, don't offer refactor.
+        if (!(flags & ModifierFlags.Export) || !wasDefault && file.symbol.exports!.has(InternalSymbolName.Default)) {
+            return undefined;
+        }
+
+        switch (exportNode.kind) {
+            case SyntaxKind.FunctionDeclaration:
+            case SyntaxKind.ClassDeclaration:
+            case SyntaxKind.InterfaceDeclaration:
+            case SyntaxKind.EnumDeclaration:
+            case SyntaxKind.ModuleDeclaration:

```

Comment:
For namespace, enum, type aliase, and variable declaration, you want to remove the `export` modifier from the declaration, and add `export default <Id>;` at the end of the file.

variable statements are gonna be tricky since they can have multiple declarations, and you can not just remove the export from them all. i would say we do not need to support variable statements with binding patterns or more than one declaration.

Reply:
Could we fix #18628 instead of working around it here?

Reply:
sure.

Reply:
Did the workaround so we can get this in for now.